Module path is not resolved properly if the name starts with android

### Issue Description

When a module is created that has a name that starts with android, ex. `androidTestModule.ts` this module is resolved as `estModule.ts`, due to a bug that is located in `packages/webpack5/src/plugins/PlatformSuffixPlugin.ts:37`.

As a result an exception is thrown:
```
WARNING in ./app/ sync \.(xml%7Cjs%7C(?<%21\.d\.)ts%7Cs?css)$ ./shared.estClass.ts
Module not found: Error: Can't resolve './shared.estClass.ts' in 'C:\Users\tizio\Workspace\Tests\testAndroidPrefixedClass\app'
 @ ./node_modules/@nativescript/webpack/dist/stubs/virtual-entry-typescript.js 4:16-103
```

Expected behavior: 
`PlatformSuffixPlugin` should only remove android if it is suffixed to the module name, like `testModule.android.ts` should be renamed `testModule.ts`


I will make a pull request in a few minutes.

### Reproduction

1. Create a brand new Nativescript project, and select the plain typescript type
2. The template is irrelevant, HelloWorld should do fine.
3. Create a new typescript class under `app/shared/androidTestClass.ts`
```
export class AndroidTestClass {
  public static Test = "123";
}
```
4. In main-page.ts add a console.log like this:
```
import { EventData, Page } from "@nativescript/core";
import { HelloWorldModel } from "./main-view-model";
import { AnddroidTestClass } from "./shared/androidTestClass";

export function navigatingTo(args: EventData) {
  const page = <Page>args.object;

  console.log(AndroidTestClass.Test); // <---- this
  page.bindingContext = new HelloWorldModel();
}
```
5. Build the app using `ns build android`


### Relevant log output (if applicable)

_No response_

### Environment

```yaml
OS: Windows 10 10.0.22621
CPU: (16) x64 Intel(R) Core(TM) i7-10870H CPU @ 2.20GHz
Shell: Not Found
node: 18.12.1
npm: 8.19.2
nativescript: Not Found

# android
java: Not Found
ndk: 21.3.6528147
apis: 28, 29, 30, 31, 32, 33
build_tools: 30.0.2, 30.0.3, 31.0.0, 32.0.0, 33.0.0
system_images:
  - android-28 | Android TV Intel x86 Atom
  - android-28 | Wear OS Intel x86 Atom
  - android-28 | Google Play Intel x86 Atom
  - android-29 | Google Play Intel x86 Atom
  - android-30 | Google Play Intel x86 Atom
  - android-31 | Google APIs Intel x86 Atom_64
  - android-33 | Google APIs Intel x86 Atom_64

# ios
xcode: Not Found
cocoapods: Not Found
python: Not Found
python3: Not Found
ruby: Not Found
platforms: Not Found
```

### Dependencies

```json
"dependencies": {
  "@nativescript/core": "~8.4.0",
  "@nativescript/theme": "~3.0.2"
},
"devDependencies": {
  "@nativescript/android": "8.4.0",
  "@nativescript/types": "~8.4.0",
  "@nativescript/webpack": "~5.0.12",
  "nativescript": "^8.4.0",
  "typescript": "~4.8.4"
}
```

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

--- a/packages/webpack5/src/plugins/PlatformSuffixPlugin.ts
+++ b/packages/webpack5/src/plugins/PlatformSuffixPlugin.ts
@@ -34,7 +34,7 @@ export class PlatformSuffixPlugin {
 	}
 
 	apply(compiler: any) {
-		const platformRE = new RegExp(`\.${this.platform}\.`);
+		const platformRE = new RegExp(`\\.${this.platform}\\.`);
 
 		// require.context
 		compiler.hooks.contextModuleFactory.tap(id, (cmf) => {

